### DISCUSSION TITLE
Escaping Double Curly Brackets in Shell Commands

### DISCUSSION THREAD

**Discussion by danloveg** [xonsh/xonsh#4876] [Q&A] [answered]
Hi, I am a big fan of Xonsh, but I'm having trouble using double curly brackets specifically in `find` commands in the shell when using the `-exec` option. I can't find anything in the documentation about how to escape this sequence of characters:
```
{}
```

Take for example this command:

```
find . -type f -exec du -sh {} \;
```

In the shell, this would run `du -sh` for each file found in the command. When I try to execute this in xonsh, the shell interpreter thinks that this is the start of a bigger expression, so starts to continue on the next line. After I hit enter in the shell, I get this output (the command doesn't actually execute):

```
/tmp $ find . -type f -exec du -sh {} \;
......
```

Here's what I've tried to get the command to execute:

```
find . -type f -exec du -sh \{\} \;
find . -type f -exec du -sh {{}} \;
$[find . -type f -exec du -sh {} \;]

# This one does execute, but find couldn't find the {}
find . -type f -exec du -sh "{}" \;
```

Anyone know what sequence of characters I can use to escape `{}`? If this would fit better as an issue lemme know!

**Comment by gforsyth** [comment, marked as answer]
Yeah, this one is a bit weird -- generally you want to escape the semi-colon using quotes, but here you also want to pass in the `{}` as a raw-string, so
```
find . -type f -exec du -sh r"{}" ";"
```

**Reply by danloveg** [reply]
Hah, that works! I was banging my head against that for so long. Appreciate it!

**Reply by anki-code** [reply]
@danloveg you can try [xontrib-sh](https://github.com/anki-code/xontrib-sh) and it will be cool to read [xonsh-cheatsheet](https://github.com/anki-code/xonsh-cheatsheet)
